cmake/sca/eclair/sca.cmake

```diff
@@ -0,0 +1,139 @@
+set(AR_ALIASES "${CMAKE_ASM_COMPILER_AR} ${CMAKE_C_COMPILER_AR} ${CMAKE_CXX_COMPILER_AR}")
+
+set(ECLAIR_ENV_ADDITIONAL_OPTIONS "")
+set(ECLAIR_REPORT_ADDITIONAL_OPTIONS "")
+
+# Default value
+set(ECLAIR_RULESET first_analysis)
+
+# ECLAIR env
+if(ECLAIR_RULESET_FIRST_ANALYSIS)
+  set(ECLAIR_RULESET first_analysis)
+elseif(ECLAIR_RULESET_STU)
+  set(ECLAIR_RULESET STU)
+elseif(ECLAIR_RULESET_STU_HEAVY)
+  set(ECLAIR_RULESET STU_heavy)
+elseif(ECLAIR_RULESET_WP)
+  set(ECLAIR_RULESET WP)
+elseif(ECLAIR_RULESET_STD_LIB)
+  set(ECLAIR_RULESET std_lib)
+elseif(ECLAIR_RULESET_USER)
+  set(ECLAIR_RULESET ${ECLAIR_USER_RULESET_NAME})
+  if(IS_ABSOLUTE ${ECLAIR_USER_RULESET_PATH})
+      set(ECLAIR_ANALYSIS_ECL_DIR ${ECLAIR_USER_RULESET_PATH})
+  else()
+    set(ECLAIR_ANALYSIS_ECL_DIR ${APPLICATION_CONFIG_DIR}/${ECLAIR_USER_RULESET_PATH})
+  endif()
+endif()
+
+# ECLAIR report
+if (ECLAIR_METRICS_TAB)
+  list(APPEND ECLAIR_REPORT_ADDITIONAL_OPTIONS "-metrics_tab=${ECLAIR_OUTPUT_DIR}/metrics")
+endif()
+if (ECLAIR_REPORTS_TAB)
+  list(APPEND ECLAIR_REPORT_ADDITIONAL_OPTIONS "-reports_tab=${ECLAIR_OUTPUT_DIR}/reports")
+endif()
+if (ECLAIR_REPORTS_SARIF)
+  list(APPEND ECLAIR_REPORT_ADDITIONAL_OPTIONS "-reports_sarif=${ECLAIR_OUTPUT_DIR}/reports.sarif")
+endif()
+if (ECLAIR_SUMMARY_TXT)
+  list(APPEND ECLAIR_REPORT_ADDITIONAL_OPTIONS "-summary_txt=${ECLAIR_OUTPUT_DIR}/summary_txt")
+endif()
+if (ECLAIR_SUMMARY_DOC)
+  list(APPEND ECLAIR_REPORT_ADDITIONAL_OPTIONS "-summary_doc=${ECLAIR_OUTPUT_DIR}/summary_doc")
+endif()
+if (ECLAIR_SUMMARY_ODT)
+  list(APPEND ECLAIR_REPORT_ADDITIONAL_OPTIONS "-summary_odt=${ECLAIR_OUTPUT_DIR}/summary_odt")
+endif()
+if (ECLAIR_FULL_TXT_ALL_AREAS)
+  list(APPEND ECLAIR_REPORT_ADDITIONAL_OPTIONS "-setq=report_areas,areas")
+endif()
+if (ECLAIR_FULL_TXT_FIRST_AREA)
+  list(APPEND ECLAIR_REPORT_ADDITIONAL_OPTIONS "-setq=report_areas,first_area")
+endif()
+if (ECLAIR_FULL_TXT)
+  list(APPEND ECLAIR_REPORT_ADDITIONAL_OPTIONS "-full_txt=${ECLAIR_OUTPUT_DIR}/report_full_txt")
+endif()
+if (ECLAIR_FULL_DOC_ALL_AREAS)
+  list(APPEND ECLAIR_REPORT_ADDITIONAL_OPTIONS "-setq=report_areas,areas")
+endif()
+if (ECLAIR_FULL_DOC_FIRST_AREA)
+  list(APPEND ECLAIR_REPORT_ADDITIONAL_OPTIONS "-setq=report_areas,first_area")
+endif()
+if (ECLAIR_FULL_DOC)
+  list(APPEND ECLAIR_REPORT_ADDITIONAL_OPTIONS "-full_doc=${ECLAIR_OUTPUT_DIR}/report_full_doc")
+endif()
+if (ECLAIR_FULL_ODT)
+  list(APPEND ECLAIR_REPORT_ADDITIONAL_OPTIONS "-full_odt=${ECLAIR_OUTPUT_DIR}/report_full_odt")
+endif()
+
+message(STATUS "ECLAIR outputs have been written to: ${ECLAIR_OUTPUT_DIR}")
+message(STATUS "ECLAIR ECB files have been written to: ${ECLAIR_ANALYSIS_DATA_DIR}")
+
+execute_process(COMMAND ${CMAKE_COMMAND} -E remove_directory ${ECLAIR_OUTPUT_DIR}
+  COMMAND ${CMAKE_COMMAND} -E make_directory ${ECLAIR_OUTPUT_DIR}
+  COMMAND ${CMAKE_COMMAND} -E make_directory ${ECLAIR_ANALYSIS_DATA_DIR}
+  COMMAND_ERROR_IS_FATAL ANY
+  COMMAND_ECHO STDOUT
+)
```

Comment:
A bit odd to have this at configure time, are you sure you want to remove the output directory like this?

Reply:
I will also use add_custom_target to delete it not in configure time, also only the Analysis data need to be deleted because we can do an incremental analysis and new ones would be overwritten / old one will be deleted.